e2etests/node_scan_rhcos_test.go

```diff
@@ -0,0 +1,187 @@
+				{
+					Id:        int64(2),
+					Name:      "tar",
+					Namespace: "rhel:8",
+					Version:   "1.27.1.el8",
+					Arch:      "x86_64",
+					Module:    "",
+					Cpes:      cpes,
+					AddedBy:   "",
+				},
+				{
+					Id:        int64(3),
+					Name:      "grep",
+					Namespace: "rhel:8",
+					Version:   "3.1-6.el8",
+					Arch:      "x86_64",
+					Module:    "",
+					Cpes:      cpes,
+					AddedBy:   "",
+				},
+			},
+			LanguageComponents: nil,
+		},
+	}
+}
+
+func TestGRPCGetRHCOSNodeVulnerabilities(t *testing.T) {
+	conn := connectToScanner(t)
+	client := v1.NewNodeScanServiceClient(conn)
+
+	cases := []struct {
+		name             string
+		request          *v1.GetNodeVulnerabilitiesRequest
+		responseContains *v1.GetNodeVulnerabilitiesResponse
+	}{
+		{
+			name:    "Selected vulnerabilities should be returned by the certified scan",
+			request: buildRequestCase([]v1.Note{}),
+			responseContains: &v1.GetNodeVulnerabilitiesResponse{
+				// We conduct a spot-checking here - more vulns can be returned from scanner for libksba and tar,
+				// but we care only about the selected one as it is sufficient for this test case.
+				// (We do not test that the set of vulns is complete, we test that the API returns any vulns if expected)
+				Features: []*v1.Feature{
+					{
+						Name:            "libksba",
+						Version:         "1.3.5-7.el8.x86_64",
+						Vulnerabilities: []*v1.Vulnerability{vulnLibksba},
+					},
+					{
+						Name:            "tar",
+						Version:         "1.27.1.el8.x86_64",
+						Vulnerabilities: []*v1.Vulnerability{vulnTar},
+					},
+					{
+						Name:            "grep",
+						Version:         "3.1-6.el8.x86_64",
+						Vulnerabilities: []*v1.Vulnerability{},
+					},
+				},
+				NodeNotes: nil,
+			},
+		},
+		{
+			name:    "Uncertified scan is unsupported for RHCOS and returns no features",
+			request: buildRequestCase([]v1.Note{v1.Note_CERTIFIED_RHEL_SCAN_UNAVAILABLE}),
+			responseContains: &v1.GetNodeVulnerabilitiesResponse{
+				Features:  []*v1.Feature{},
+				NodeNotes: nil,
+			},
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			c := c
+			resp, err := client.GetNodeVulnerabilities(context.Background(), c.request)
+			require.NoError(t, err)
+			assert.Len(t, resp.GetFeatures(), len(c.responseContains.GetFeatures()))
+			for _, expectedFeat := range c.responseContains.GetFeatures() {
+				var feat *v1.Feature
+				for _, gotFeat := range resp.GetFeatures() {
+					if expectedFeat.GetName() == gotFeat.GetName() && expectedFeat.GetVersion() == gotFeat.GetVersion() {
+						feat = gotFeat
+					}
+				}
+				assert.NotNil(t, feat, "expected to find feat '%s:%s' in the reply, but got none. Features in the reply: %+v", expectedFeat.GetName(), expectedFeat.GetVersion(), resp.GetFeatures())
+				assertIsSubset(t, feat.GetVulnerabilities(), expectedFeat.GetVulnerabilities())
+			}
+			assert.Equal(t, c.responseContains.GetNodeNotes(), resp.GetNodeNotes())
```

Comment:
I believe it's meant to be `assert.Equal(t, expected, actual)`

Reply:
Yes, this is correct. I tried to unify this a bit and used `assert.Len` (which uses different order). PTAL at  [6692db4](https://github.com/stackrox/scanner/pull/1004/commits/6692db429eba7737b9c090e679f5fe5dad836a39) and let me know what you think.